File: services/shared/Shared.Common/Domain/AggregatesModels/Book.cs
namespace ECommerce.Shared.Common.Domain.AggregatesModels;
public class Book : BaseEntity
{
    public int NumberOfPages { get; set; } = 0;
    public List<string> Contributions { get; set; } = new List<string>();
    public string? PublishCountry { get; set; }
    public string? PublishDate { get; set; }
    public List<string> Series { get; set; } = new List<string>();
    public List<string> Publishers { get; set; } = new List<string>();
    public List<string> Isbn13 { get; set; } = new List<string>();
    public List<string> Isbn10 { get; set; } = new List<string>();
    public BookStatus Status { get; private set; } = BookStatus.OTHER;
    public void UpdateStatus(BookStatus status)
    {
        Status = status;
    }
}

public enum BookStatus
{
    BORROW,
    FULL,
    RESTRICTED,
    NOVIEW,
    OTHER
}

File: services/bookstore-web/src/models/book.ts
import { RefType, BookStatus } from "./base";

export type SearchProductItem = {
  id: string;
  title: string;
  refType: RefType;

  lendingBookId?: string;
  lendingBookStatus?: BookStatus;

  coverBookId: string;
  coverImageUrl?: string;

  hasFullText: boolean;

  wantToReadCount: number;
  alreadyReadCount: number;

  relatedBookImgs: string[];

  firstPublishYear?: number;
  authors: IdName[];

  ratingsAverage: number;
  ratingsCount: number;

  firstEditionId?: string;
  editionCount: number;
};

export type Entity = {
  id: string;
  title: string;
  subtitle?: string;
  description?: string;

  imageUrlS?: string;
  imageUrlM?: string;

  refType: RefType;

  coverImages: string[];
};

export type Book = Entity & {
  numberOfPages: number;

  status: BookStatus;

  publishCountry?: string;
  publishDate?: string;
  contributions: string[];
  series: string[];
  publishers: string[];
  isbn13: string[];
  isbn10: string[];

  workId?: string;
};

export type Work = Entity & {
  subjects?: string[];
  subjectPeople?: string[];
  subjectTimes?: string[];
  subjectPlaces?: string[];
};

export type WorkRatings = {
  average: number;
  count: number;
  rating1Stars: number;
  rating2Stars: number;
  rating3Stars: number;
  rating4Stars: number;
  rating5Stars: number;
};
